src/sys/termios.rs

```diff
@@ -1,648 +1,782 @@
+//! An interface for controlling asynchronous communication ports
+
 use {Errno, Result};
-use libc::c_int;
+use libc::{self, c_int, pid_t, tcflag_t};
 use std::mem;
 use std::os::unix::io::RawFd;
 
-pub use self::ffi::consts::*;
-pub use self::ffi::consts::SetArg::*;
-pub use self::ffi::consts::FlushArg::*;
-pub use self::ffi::consts::FlowArg::*;
-
-mod ffi {
-    pub use self::consts::*;
-
-    #[cfg(any(target_os = "macos", target_os = "freebsd", target_os = "dragonfly", target_os = "openbsd", target_os = "netbsd", target_os = "linux"))]
-    mod non_android {
-        use super::consts::*;
-        use libc::c_int;
-
-        // `Termios` contains bitflags which are not considered
-        // `foreign-function-safe` by the compiler.
-        #[allow(improper_ctypes)]
-        extern {
-            pub fn cfgetispeed(termios: *const Termios) -> speed_t;
-            pub fn cfgetospeed(termios: *const Termios) -> speed_t;
-            pub fn cfsetispeed(termios: *mut Termios, speed: speed_t) -> c_int;
-            pub fn cfsetospeed(termios: *mut Termios, speed: speed_t) -> c_int;
-            pub fn tcgetattr(fd: c_int, termios: *mut Termios) -> c_int;
-            pub fn tcsetattr(fd: c_int,
-                             optional_actions: c_int,
-                             termios: *const Termios) -> c_int;
-            pub fn tcdrain(fd: c_int) -> c_int;
-            pub fn tcflow(fd: c_int, action: c_int) -> c_int;
-            pub fn tcflush(fd: c_int, action: c_int) -> c_int;
-            pub fn tcsendbreak(fd: c_int, duration: c_int) -> c_int;
-        }
-    }
+/// FIXME: This struct has no documentation within libc, which is very obvious when generating
+///        docs
+pub use libc::termios as Termios;
+
+/// Baud rates supported by the system
+///
+/// B0 is special and will disable the port.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum BaudRate {
```

Comment:
These enums are going to have different values than the similarly named constants from libc, so implementing the `From` trait below is going to require real work at runtime.  Did you consider simply wrapping libc's constants in a enum?  That way, converting back and forth can be done by simple casts.  See `nix::sys::Signal` or `nix::sys::LioOpcode` for examples.